cgen error: json.encode fails on type alias

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 18.04.6 LTS (VM)
Processor: 2 cpus, 64bit, little endian, AMD EPYC 7551 32-Core Processor
CC version: cc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0

getwd: /home/ubuntu/workspace/im_server
vmodules: /home/ubuntu/.vmodules
vroot: /home/ubuntu/v
vexe: /home/ubuntu/v/v
vexe mtime: 2022-08-21 19:24:28
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.0 c356882.0d9ac1f

Git version: git version 2.17.1
Git vroot status: weekly.2022.33-46-g0d9ac1f5
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 827f7452

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg json.v`
```v
import json

struct Aa {
	sub		SumType
}

struct Bb {
	a		int
}

type SumType = Bb

println(json.encode(Aa{
	sub: Bb{
		a: 1
	}
}))
```

**What did you expect to see?**

The object should be encoded corretly


**What did you see instead?**
```
==================
/tmp/v_1001/json.3433979148997518964.tmp.c:2176: warning: cast between pointer and integer of different size
tcc: error: undefined symbol 'json__encode_main__Bb'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/json/json_test.v
+++ b/vlib/json/json_test.v
@@ -488,3 +488,23 @@ struct StByteArray {
 fn test_byte_array() {
 	assert json.encode(StByteArray{ ba: [byte(1), 2, 3, 4, 5] }) == '{"ba":[1,2,3,4,5]}'
 }
+
+struct Aa {
+	sub SumType
+}
+
+struct Bb {
+	a int
+}
+
+type SumType = Bb
+
+fn test_encode_alias_field() {
+	s := json.encode(Aa{
+		sub: SumType(Bb{
+			a: 1
+		})
+	})
+	println(s)
+	assert s == '{"sub":{"a":1}}'
+}

--- a/vlib/v/gen/c/json.v
+++ b/vlib/v/gen/c/json.v
@@ -436,7 +436,7 @@ fn (mut g Gen) gen_struct_enc_dec(type_info ast.TypeInfo, styp string, mut enc s
 					}
 					dec.writeln('\t}')
 				} else {
-					g.gen_json_for_type(field.typ)
+					g.gen_json_for_type(alias.parent_type)
 					tmp := g.new_tmp_var()
 					gen_js_get_opt(dec_name, field_type, styp, tmp, name, mut dec, is_required)
 					dec.writeln('\tif (jsonroot_$tmp) {')

